### PR TITLE
For suggestions first try matching both tags

### PR DESCRIPTION
**Pull request by psoleckimoj**
Currently we try to match items for each secondary tag then each primary tag. First try should be to find anything that matches both secondary tags

- Update API with new method for matching both
- Try new method first before moving onto existing matching

### CODE REVIEW

File: moj-be/modules/custom/moj_resources/src/SuggestedContentApiClass.php [hunk 1]
```diff
@@ -97,16 +97,38 @@ private function translateNode(NodeInterface $node)
...
     }
 
     return array_slice($detailed_results, 0, $number);
   }
 
+  private function getSecondaryTagIds($secondary_tags) {
+    $tag_ids = [];
+    $number_secondary_tags = count($secondary_tags);
+
+    for ($i = 0; $i < $number_secondary_tags; $i++) {
```
Comment (romidane) [inline moj-be/modules/custom/moj_resources/src/SuggestedContentApiClass.php]: Have a look at

File: moj-be/modules/custom/moj_resources/src/SuggestedContentApiClass.php [hunk 2]
```diff
@@ -97,16 +97,38 @@ private function translateNode(NodeInterface $node)
   private function getSuggestions($nid, $number, $prison)
   {
     $node = $this->node_storage->load($nid);
-    $detailed_results = $this->combineSecondaryTagItems($node->field_moj_secondary_tags, $number, $prison);
+    $secondary_tags = $node->field_moj_secondary_tags;
+
+    $detailed_results = $this->getBothSecondaryTagItemsFor(
```
Comment (romidane) [inline moj-be/modules/custom/moj_resources/src/SuggestedContentApiClass.php]: `Both` is a little too specific I think `getSecondaryTagItemsFor` will be fine

Comment (psoleckimoj) [reply moj-be/modules/custom/moj_resources/src/SuggestedContentApiClass.php]: Done

File: moj-be/modules/custom/moj_resources/src/SuggestedContentApiClass.php [hunk 3]
```diff
@@ -97,16 +97,38 @@ private function translateNode(NodeInterface $node)
...
+      $this->getSecondaryTagIds($secondary_tags),
+      $number,
+      $prison
+    );
 
     if (count($detailed_results) < $number) {
-      $combined_primary_tag_items = $this->combinePrimaryTagItems($node->field_moj_categories, $number, $prison);
-      $detailed_results = array_merge($combined_primary_tag_items, $detailed_results);
+      $combined_secondary_tag_items = $this->combineSecondaryTagItems($secondary_tags, $number, $prison);
```
Comment (romidane) [inline moj-be/modules/custom/moj_resources/src/SuggestedContentApiClass.php]: I don't think you need to nest these if statements?

**Pull request merged**